conn.go

```diff
@@ -666,6 +671,22 @@ func (cn *conn) simpleQuery(q string) (res *rows, err error) {
 	}
 }
 
+type noRows struct{}
+
+var emptyRows noRows
+
+var _ driver.Result = noRows{}
+
+var ErrNoRowsAffected = errors.New("no RowsAffected available after the empty statement")
```

Comment:
Another way to do this is to define a public method that a user defined interface can be asserted against (for those users who really need to detect this error).